--- role_include.py.orig	2017-05-19 07:50:06.405443799 +0800
+++ role_include.py	2017-05-19 07:50:54.252532305 +0800
@@ -26,6 +26,7 @@
 from ansible.playbook.task import Task
 from ansible.playbook.role import Role
 from ansible.playbook.role.include import RoleInclude
+from ansible.template import Templar
 
 try:
     from __main__ import display
@@ -68,6 +69,9 @@
             myplay =  self._parent._play
         else:
             myplay = play
+        templar = Templar(loader=loader, variables=self.vars)
+        if templar.templatable(self._role_name):
+            self._role_name = templar.template(self._role_name)
 
         ri = RoleInclude.load(self._role_name, play=myplay, variable_manager=variable_manager, loader=loader)
         ri.vars.update(self.vars)
