--- task_executor.py.orig	2017-05-19 07:46:13.237884744 +0800
+++ task_executor.py	2017-05-19 07:46:27.417614773 +0800
@@ -456,6 +456,7 @@
         # if this task is a IncludeRole, we just return now with a success code so the main thread can expand the task list for the given host
         elif self._task.action == 'include_role':
             include_variables = self._task.args.copy()
+            include_variables = templar.template(include_variables)
             return dict(include_role=self._task, include_variables=include_variables)
 
         # Now we do final validation on the task, which sets all fields to their final values.
